pennylane/ops/op_math/controlled_class.py

```diff
@@ -177,6 +180,10 @@ def __init__(
                     "Specifying control values as a string is deprecated. Please use Sequence[Bool]",
                     UserWarning,
                 )
+
+                if not set(control_values).issubset({"0", "1"}):
+                    raise ValueError("String of control values can contain only '0' or '1'.")
+
                 control_values = [(x == "1") for x in control_values]
```

Comment:
Again the eternal discussion of whether to have a lot of input checks and less performance, or give more freedom to the user to improve performance 😆 

In this case I would not add this check, given that in python casting to bool any number different than 0 returns `True`:
```pycon
>>> bool(2)
True
```

Thus we can do the following to keep with python standards:
```suggestion
                control_values = [(x != "0") for x in control_values]
```